exp/addons/api/v1beta1/clusterresourcesetbinding_types.go

```diff
@@ -129,6 +129,10 @@ type ClusterResourceSetBindingSpec struct {
 	// Bindings is a list of ClusterResourceSets and their resources.
 	// +optional
 	Bindings []*ResourceSetBinding `json:"bindings,omitempty"`
+
+	// ClusterName is the name of the Cluster this binding applies to.
+	// +optional
+	ClusterName *string `json:"clusterName,omitempty"`
```

Comment:
Why is this a pointer and optional?

Looks like in other APIs we usually define this as:
```go
	// ClusterName is the name of the Cluster this object belongs to.
	ClusterName string `json:"clusterName"`
```

Reply:
Is this because we are introducing a new field to the API to keep it downdwards-compatible?

If yes, can we open an issue to clean this up in v1beta2?

Reply:
Sorry for the back and forth @chaunceyjiang  - @sbueringer is 100 percent correct, this field should match with the fields in other API objects.

Reply:
Wait. I think it's more complicated :) 

1. We probably cannot introduce a new mandatory field without bumping the apiVersion. So we can introduce it now as optional and then make it mandatory with v1beta2?

2. Let's assume we want to add this field as optional. 
> An optional field MUST be marked with +optional and include an omitempty JSON tag.
> Fields SHOULD be pointers if there is a good reason for it, for example:
>   the nil and the zero values (by Go standards) have semantic differences.
>   Note: This doesn’t apply to map or slice types as they are assignable to nil.
>   the field is of a struct type, contains only fields with omitempty and you want to prevent that it shows up as an empty > object after marshalling (e.g. kubectl get)

So I think the field MUST have +optional and omitempty. I think it doesn't have to be a pointer. There is no relevant difference for us between ClusterName being empty or not being set at all, right?

Reply:
Fair enough - there's no difference here between nil and empty in this case.

Reply:
Okay let's go with the following for now:
```go
	// ClusterName is the name of the Cluster this binding applies to.
	// +optional
	ClusterName string `json:"clusterName,omitempty"`
```

I think before v1beta2 we should do some experiments, especially with new'ish kubectl versions (OpenAPi schema validation will be dropped at some point on client-side) but also in general. Then probably re-write our API conventions and cleanup all our APIs. But let's go with our current policy for now. That should be okay